exploration/code-mode-introducer.md

```diff
@@ -0,0 +1,224 @@
+# Design Proposal: Choosing a Code Mode Introducer
+
+Status: **Proposed**
+
+<details>
+	<summary>Metadata</summary>
+	<dl>
+		<dt>Contributors</dt>
+		<dd>@aphillips</dd>
+		<dt>First proposed</dt>
+		<dd>2023-11-10</dd>
+		<dt>Pull Requests</dt>
+		<dd>#000</dd>
+	</dl>
+</details>
+
+## Objective
+
+_What is this proposal trying to achieve?_
+
+It must be possible to reliably parse messages.
+
+Our current syntax features unquoted patterns for simple messages
+and unquoted code tokens with quoted patterns for complex messages.
+Determining whether a message will have code tokens requires some
+special character sequence, either part of the code itself or
+prepended to the message.
+This proposal examines the options for determining code mode.
+
+## Background
+
+_What context is helpful to understand this proposal?_
+
+## Use-Cases
+
+_What use-cases do we see? Ideally, quote concrete examples._
+
+As a developer, I want to create messages with the minimal amount of special syntax.
+I don't want to have to type additional characters that add no value.
+I want the syntax to be logical and as consistent as possible.
+
+As a translator, I don't want to have to learn special syntax to support features such as declarations.
+
+As a user, I want my messages to be robust.
+Minor edits and changes should not result in syntax errors.
+
+As a user, I want to be able to see which messages are complex at a glance
+and to parse messages into their component parts visually as easily as possible.
+
+## Requirements
+
+_What properties does the solution have to manifest to enable the use-cases above?_
+
+## Constraints
+
+_What prior decisions and existing conditions limit the possible design?_
+
+Some of the options use a new sigil as part of the introducer.
+For various reasons, `#` has been used recently as a placeholder for this sigil.
+There are concerns that this character is not suitable, since it is used as a comment
+introducer in a number of formats. 
+See for example #520.
+The actual sigil used needs to be an ASCII character in the reserved or private use
+set (with syntax adjustments if we use up a private-use one).
+Most of the options below have been changed to use `^`, using 
+Apple's experimental syntax as a model for sigil choice.
```

Comment:
Not sure what Apple's experimental syntax is -- can you add a link?

Reply:
I got this from @grhoten's presentation at the Unicode Technical Workshop this past week. I don't have a link handy, but one will be in the offing I think.

Reply:
I remember watching https://developer.apple.com/videos/play/wwdc2023/10153/ which demoed a syntax which used `^`: `^[Bienvenido](inflect: true) a tu iPhone`.